core/iwasm/common/wasm_shared_memory.c

```diff
@@ -308,7 +313,7 @@ acquire_wait_info(void *address, bool create)
     if (address)
         wait_info = (AtomicWaitInfo *)bh_hash_map_find(wait_map, address);
 
-    if (!create) {
+    if (!wait_node && !wait_info) {
```

Comment:
Not sure why need to add `&& !wait_info`? That means we will create the wait_info even when this function is called from `wasm_runtime_atomic_notify`, not sure whether it is necessary?

Reply:
Fixed